src/utils/assets.js

```diff
@@ -151,11 +152,34 @@ export const addressesEqual = (address1: ?string, address2: ?string): boolean =>
 };
 
 export const getAssetData = (
-  assetsData: Asset[],
+  userAssets: Asset[],
   supportedAssetsData: Asset[],
   assetSymbol: string,
 ): Asset | Object => {
-  return assetsData.find(({ symbol }: Asset) => symbol === assetSymbol)
+  return userAssets.find(({ symbol }: Asset) => symbol === assetSymbol)
   || supportedAssetsData.find(({ symbol }: Asset) => symbol === assetSymbol)
   || {};
 };
+
+export const getAssetDataByAddress = (
+  userAssets: Asset[],
+  supportedAssetsData: Asset[],
+  assetAddress: string,
+): Asset | Object => {
+  return userAssets.find(({ address }: Asset) => isCaseInsensitiveMatch(address, assetAddress))
+  || supportedAssetsData.find(({ address }: Asset) => isCaseInsensitiveMatch(address, assetAddress))
+  || {};
```

Comment:
As a suggestion, I'd make the return value of this function optional and then make the caller check the result, since making this return `Asset | Object` is almost the same as making it to be just `Object`.

We could write this one as:

```js
const getAssetDataByAddress = (
  userAssets: Asset[],
  supportedAssetsData: Asset[],
  assetAddress: string,
): ?Asset => {
  return userAssets.find(({ address }: Asset) => isCaseInsensitiveMatch(address, assetAddress))
  || supportedAssetsData.find(({ address }: Asset) => isCaseInsensitiveMatch(address, assetAddress));
};
```

And then the caller could be something like:

```js
export const getAssetSymbolByAddress = (assets: Asset[], supportedAssets: Asset[], address: ?string): ?string => {
  if (!address) return;

  // NOTE: ZERO_ADDRESS usually means it's ETH transaction
  if (address === ZERO_ADDRESS) return ETH;

  const assetData = getAssetDataByAddress(assets, supportedAssets, address);
  if (assetData) return assetData.symbol;
};
```

Reply:
Wanted this method of `getAssetDataByAddress` to have the same approach as `getAssetData` as it has same purpose while for `getAssetDataByAddress` we are destructuring right away from method in many places of the code, if you OK changing that then we might need to revisit all places where it's used and apply the removal of destructuring from method.

Reply:
Looks like it's better to do the suggested refactoring in a new PR